streams/src/test/java/org/apache/kafka/streams/tests/StreamsUpgradeTest.java

```diff
@@ -116,10 +122,25 @@ public static void main(final String[] args) throws Exception {
 
     public static class FutureStreamsPartitionAssignor extends StreamsPartitionAssignor {
 
+        private AtomicInteger usedSubscriptionMetadataVersionPeek;
+
         public FutureStreamsPartitionAssignor() {
             usedSubscriptionMetadataVersion = LATEST_SUPPORTED_VERSION + 1;
         }
 
+        @Override
+        public void configure(final Map<String, ?> configs) {
```

Comment:
Why this change? What's `usedSubscriptionMetadataVersionPeek` used for? I don't see it in any assertions or something along those lines.

Reply:
Oh, it's used in `StreamsUpgradeTestIntegrationTest`, but not via field access. You can pass in an AtomicInteger, and then use it in assertions to "peek" at the current value for `usedSubscriptionMetadataVersion` internal to the assignor.

I'm using it to make direct assertions about the version at different points in the protocol without having to grep the logs.